frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -366,7 +366,9 @@ export const tabsReducer = createReducer<DashboardState>(
       ) => {
         const dashCard = state.dashcards[dashCardId];
 
-        dashCard.row = originalRow;
+        // when a card is moved, another card could steal its position,
+        // this trick gives the original tab precedence on the position
+        dashCard.row = originalRow - 0.1;
```

Comment:
Let's put this magic number into a constant (not sure about naming, maybe UNDO_ROW_PRECEDENCE or something)

Reply:
Thanks for the suggestion, I moved it to a function so that I could also consolidate the comment and explanation in just one place.
Hopefully this also makes the solution a bit less hacky